src/kcas.ml

```diff
@@ -3,7 +3,17 @@
  * Copyright (c) 2023, Vesa Karvonen <vesa.a.j.k@gmail.com>
  *)
 
+(*
+external fenceless_get : 'a Atomic.t -> 'a = "%field0"
+external fenceless_set : 'a Atomic.t -> 'a -> unit = "%setfield0"
+*)
+(**)
+let fenceless_get = Atomic.get
+let fenceless_set = Atomic.set
```

Comment:
I plan to add a PR that changes these aliases to actually perform fenceless operations.  Having these as fenceless should be safe (because the fences would be redundant), seems to significantly improve performance on ARM (Apple M1), and is also completely internal to the library, so I personally feel that we should just make the optimization.